web/src/app/util/AppBarSearchContainer.js

```diff
@@ -14,94 +14,85 @@ export function SearchContainer() {
   )
 }
 
-export class SearchProvider extends React.PureComponent {
-  state = {
-    actions: null,
-  }
+export function SearchProvider(props) {
+  const [actions, setActions] = useState(null)
 
-  _mountCount = 0
-  _mounted = false
+  let _mountCount = 0
+  let _mounted = false
+  let _pending = null
 
-  componentDidMount() {
-    this._mounted = true
-    if (this._pending) {
-      this.setActions(this._pending)
-      this._pending = null
+  function _handleSetActions(actns) {
+    if (!_mounted) {
+      _pending = actns
+      return
     }
-  }
 
-  componentWillUnmount() {
-    this._mounted = false
-    this._pending = false
-    this.setActions.cancel()
+    setActions(actns)
   }
 
-  _setActions = (actions) => {
-    if (!this._mounted) {
-      this._pending = actions
-      return
-    }
+  const handleSetActions = debounce(_handleSetActions)
 
-    this.setState({ actions })
-  }
+  useEffect(() => {
+    _mounted = true
+    if (_pending) {
+      handleSetActions(_pending)
+      _pending = null
+    }
 
-  setActions = debounce(this._setActions)
+    return () => {
+      _mounted = false
+      _pending = false
+      handleSetActions.cancel()
+    }
+  }, [])
 
-  updateMounted = (mount) => {
+  function updateMounted(mount) {
     if (mount) {
-      this._mountCount++
+      _mountCount++
     } else {
-      this._mountCount--
+      _mountCount--
     }
 
-    if (this._mountCount > 1 && global.console && console.error) {
+    if (_mountCount > 1 && global.console && console.error) {
       console.error(
         'Search: Found more than one <AppBarSearchContainer> component mounted within the same provider.',
       )
     }
   }
 
-  render() {
-    return (
-      <SearchContext.Provider
-        value={{
-          actions: this.state.actions,
-          setActions: this.setActions,
-          trackMount: this.updateMounted,
-        }}
-      >
-        {this.props.children}
-      </SearchContext.Provider>
-    )
-  }
+  return (
+    <SearchContext.Provider
+      value={{
+        actions: actions,
+        setActions: props.setActions,
```

Comment:
```suggestion
        setActions: handleSetActions,
```

Reply:
Changed `props.setActions` to `handleSetActions`. Tested by dragging browser to a smaller width.